src/Raven.Studio/typescript/viewmodels/resources/widgets/ongoingTasksWidget.ts

```diff
@@ -1,202 +1,277 @@
-        this.nodes(nodeSet);
-    }
-}
-
-// inner table item
-class databaseTaskItem {
-    databaseName = ko.observable<string>();
-    taskCount = ko.observable<number>();
-    taskNodes = ko.observable<taskNodes>();
-    
-    constructor() {
-        this.taskNodes(new taskNodes());
+class rawTaskItem {
+    constructor(public type: string, public count: number, public dbName: string, public node: string) {
     }
 }
 
-// parent table item
-class ongoingTaskItem {
-    taskName = ko.observable<string>();
+class taskItem {
+    taskType = ko.observable<string>();
     taskCount = ko.observable<number>();
-    taskNodes = ko.observable<taskNodes>();
+    taskNodes = ko.observableArray<string>();
     
-    innerItems = ko.observableArray<databaseTaskItem>([]);
+    databaseName = ko.observable<string>();
 
-    typeClass = ko.observable<string>();
-    iconClass = ko.observable<string>();
+    isTitleItem = ko.observable<boolean>(false);
+    even: boolean = false;
 
-    constructor(name: string, count: number, typeClass: string, iconClass: string) {
-        this.taskName(name);
+    constructor(type: string, count: number, dbName?: string, nodes?: string[]) {
+        this.taskType(type);
+        this.databaseName(dbName);
         this.taskCount(count);
-        this.taskNodes(new taskNodes());
+        this.taskNodes(nodes);
         
-        this.typeClass(typeClass);
-        this.iconClass(iconClass);
-    }
-
-    getTaskNameForUI(): TasksNamesInUI {
-        switch (this.taskName()) {
-            case "ExternalReplicationCount": return "External Replication";
-            case "ReplicationHubCount": return "Replication Hub";
-            case "ReplicationSinkCount": return "Replication Sink";
-            case "RavenEtlCount": return "RavenDB ETL";
-            case "OlapEtlCount": return "OLAP ETL";
-            case "SqlEtlCount": return "SQL ETL";
-            case "PeriodicBackupCount": return "Backup";
-            case "SubscriptionCount": return "Subscription";
+        if (!dbName) {
+            this.isTitleItem(true);
         }
     }
+
+    updateWith(countToAdd: number, nodeToAdd: string) {
+        this.taskCount(this.taskCount() + countToAdd);
+
+        const foundTag = this.taskNodes().find(x => x === nodeToAdd);
+        if (!foundTag) {
+            this.taskNodes.push(nodeToAdd);
+        }
+    }
+    
+    static itemFromRaw(rawItem: rawTaskItem): taskItem {
+        return new taskItem(rawItem.type, rawItem.count, rawItem.dbName, [rawItem.node]);
+    }
 }
 
-class nodeRawData {
-    nodeTag = ko.observable<string>();
-    databaseItems = ko.observableArray<Raven.Server.Dashboard.DatabaseOngoingTasksInfoItem>([]);
+interface taskInfo {
+    nameForUI: string,
+    icon: string,
+    colorClass: string
 }
 
 class ongoingTasksWidget extends websocketBasedWidget<Raven.Server.Dashboard.Cluster.Notifications.OngoingTasksPayload> {
+
+    static readonly taskInfoRecord: Record<string, taskInfo> = {
```

Comment:
can we avoid string in record key?

Reply:
changed to use:
`Raven.Client.Documents.Operations.OngoingTasks.OngoingTaskType`

Reply:
nice - better then regular string :)